sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -914,6 +914,42 @@ object functions {
   def ceil(columnName: String): Column = ceil(Column(columnName))
 
   /**
+   * Computes the concat of the given values.
+   *
+   * @group normal_funcs
+   * @since 1.5.0
+   */
+  @scala.annotation.varargs
+  def concat(exprs: Column*): Column = Concat(exprs.map(_.expr): _*)
+
+  /**
+   * Computes the concat of the given column names.
+   *
+   * @group normal_funcs
+   * @since 1.5.0
+   */
+  @scala.annotation.varargs
+  def concat(columnName: String, columnNames: String*): Column =
+    concat((columnName +: columnNames).map(Column.apply): _*)
+
+  /**
+   * Computes the concat_ws of the given columns with given separator.
+   *
+   * @group normal_funcs
+   * @since 1.5.0
+   */
+  @scala.annotation.varargs
+  def concat_ws(sep: Column, exprs: Column*): Column = concat_ws(sep, array(exprs: _*))
+
+  /**
+   * Computes the concat_ws of the given columns with given separator.
+   *
+   * @group normal_funcs
+   * @since 1.5.0
+   */
+  def concat_ws(sep: Column, exprs: Column): Column = ConcatWS(sep.expr, exprs.expr)
```

Comment:
since array() didn't provide the interface with columnNames, so I just omit them for convenience, since I want to support the format of `concat_ws(sep, array(exprs))` here.